src/components/timeline/index.js

```diff
@@ -5,328 +5,33 @@
-import type { Milliseconds, StartEndRange } from '../../types/units';
 import type { ConnectedProps } from '../../utils/connect';
 
 type StateProps = {|
-  +committedRange: StartEndRange,
-  +globalTracks: GlobalTrack[],
-  +globalTrackOrder: TrackIndex[],
-  +globalTrackReferences: GlobalTrackReference[],
-  +panelLayoutGeneration: number,
-  +zeroAt: Milliseconds,
-  +timelineType: TimelineType,
-  +hiddenTrackCount: HiddenTrackCount,
-  +activeBrowsingContextID: BrowsingContextID | null,
   +showTabOnly: BrowsingContextID | null,
 |};
 
-type DispatchProps = {|
-  +changeGlobalTrackOrder: typeof changeGlobalTrackOrder,
-  +changeTimelineType: typeof changeTimelineType,
-  +changeRightClickedTrack: typeof changeRightClickedTrack,
-  +changeViewAndRecomputeProfileData: typeof changeViewAndRecomputeProfileData,
-|};
-
-type Props = {|
-  ...SizeProps,
-  ...ConnectedProps<{||}, StateProps, DispatchProps>,
-|};
-
-type State = {|
-  initialSelected: InitialSelectedTrackReference | null,
-|};
-
-class TimelineSettingsGraphType extends React.PureComponent<{|
-  +timelineType: TimelineType,
-  +changeTimelineType: typeof changeTimelineType,
-|}> {
-  _changeToCategories = () => this.props.changeTimelineType('category');
-  _changeToStacks = () => this.props.changeTimelineType('stack');
-
-  render() {
-    const { timelineType } = this.props;
-
-    return (
-      <form>
-        <div className="timelineSettingsToggle">
-          Graph type:{' '}
-          <label className="photon-label photon-label-micro timelineSettingsToggleLabel">
-            <input
-              type="radio"
-              name="timelineSettingsToggle"
-              className="photon-radio photon-radio-micro timelineSettingsToggleInput"
-              checked={timelineType === 'category'}
-              onChange={this._changeToCategories}
-            />
-            Categories
-          </label>
-          <label className="photon-label-micro timelineSettingsToggleLabel">
-            <input
-              type="radio"
-              name="timelineSettingsToggle"
-              className="photon-radio photon-radio-micro timelineSettingsToggleInput"
-              checked={timelineType === 'stack'}
-              onChange={this._changeToStacks}
-            />
-            Stack height
-          </label>
-        </div>
-      </form>
-    );
-  }
-}
-
-class TimelineSettingsHiddenTracks extends React.PureComponent<{|
-  +hiddenTrackCount: HiddenTrackCount,
-  +changeRightClickedTrack: typeof changeRightClickedTrack,
-|}> {
-  _showMenu = (event: SyntheticMouseEvent<HTMLElement>) => {
-    const rect = event.currentTarget.getBoundingClientRect();
-    this.props.changeRightClickedTrack(null);
-    showMenu({
-      data: null,
-      id: 'TimelineTrackContextMenu',
-      position: { x: rect.left, y: rect.bottom },
-      target: event.target,
-    });
-  };
+type Props = ConnectedProps<{||}, StateProps, {||}>;
 
+class Timeline extends React.PureComponent<Props> {
```

Comment:
This is the new intermediate Timeline class which redirects to either `FullTimeline` or `ActiveTabTimeline`. The diff makes it complicated because github still tries to think that I overwrote the previous index.js even though I did `git mv` in the previous commit.

Reply:
Ah, GitHub is doing the right thing with the separate commits. Thanks for doing it that way!